src/meta/processors/BaseProcessor.inl

```diff
@@ -448,9 +457,10 @@ BaseProcessor<RESP>::getIndexes(GraphSpaceID spaceId,
     auto indexIter = iterRet.value().get();
     while (indexIter->valid()) {
         auto item = MetaServiceUtils::parseIndex(indexIter->val());
-        auto id = isEdge ? item.get_schema_id().get_edge_type()
-                         : item.get_schema_id().get_tag_id();
-        if (item.get_schema_id().getType() == type && id == edgeOrTag) {
+        if (item.get_schema_id().getType() == nebula::cpp2::SchemaID::Type::tag_id &&
+            item.get_schema_id().get_tag_id() == schemaId) {
+            items.emplace_back(std::move(item));
+        } else if (item.get_schema_id().get_edge_type() == schemaId) {
             items.emplace_back(std::move(item));
         }
```

Comment:
Here is a logic error : when item is tagIndex ( tagId == 1),
Serious errors will occur when call getIndexes(spaceId, tagId == 2) , the reason is item.get_schema_id().get_edge_type() is unpredictable.